src/transformers/modeling_tf_dpr.py

```diff
@@ -0,0 +1,777 @@
+        super().__init__(config, *args, **kwargs)
+        
+        # resolve name conflict with TFBertMainLayer instead of TFBertModel
+        # self.bert_model = TFBertModel(config, name = 'bert_model')
+        self.bert_model = TFBertMainLayer(config, name = 'bert_model')
+        self.bert_model.config = config
+
+        assert self.bert_model.config.hidden_size > 0, "Encoder hidden_size can't be zero"
+        self.projection_dim = config.projection_dim
+        if self.projection_dim > 0:
+            self.encode_proj = Dense(config.projection_dim, 
+                                     kernel_initializer=get_initializer(config.initializer_range), 
+                                     name="encode_proj" 
+                                     )
+        
+        self.init_weights()
+
+    def call(
+        self,
+        input_ids: Tensor,
+        attention_mask: Optional[Tensor] = None,
+        token_type_ids: Optional[Tensor] = None,
+        inputs_embeds: Optional[Tensor] = None,
+        output_attentions: bool = False,
+        output_hidden_states: bool = False,
+        return_dict: bool = None,
+        training: bool = False,
+    ) -> Union[TFBaseModelOutputWithPooling, Tuple[Tensor, ...]]:
+        
+        return_dict = return_dict if return_dict is not None else self.bert_model.return_dict
+        
+        outputs = self.bert_model(
+            inputs=input_ids,
+            attention_mask=attention_mask,
+            token_type_ids=token_type_ids,
+            inputs_embeds=inputs_embeds,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+            training=training
+        )
+        sequence_output, pooled_output = outputs[:2]
+        pooled_output = sequence_output[:, 0, :]
+        if self.projection_dim > 0:
+            pooled_output = self.encode_proj(pooled_output)
+
+        if not return_dict:
+            return (sequence_output, pooled_output) + outputs[2:]
+
+        return TFBaseModelOutputWithPooling(
+            last_hidden_state=sequence_output,
+            pooler_output=pooled_output,
+            hidden_states=outputs.hidden_states,
+            attentions=outputs.attentions,
+        )
+
+    @property
+    def embeddings_size(self) -> int:
+        if self.projection_dim > 0:
+            return self.projection_dim
+        return self.bert_model.config.hidden_size
+
+    def init_weights(self):
+        pass
+        # Not translate as there's no tf's init_weights counterpart
+        # Note that tf already performs init_weights when creating layers
+        
+        # Pytorch's code (for reference)
+        # self.bert_model.init_weights()
+        # if self.projection_dim > 0:
+        #     self.encode_proj.apply(self.bert_model._init_weights)
+
+
+class TFDPRSpanPredictor(TFPreTrainedModel):
+
+    base_model_prefix = "span_predictor"
```

Comment:
The base model prefix should be the same as the base model defined below:
```py
        self.encoder = TFDPREncoder(config, name = 'encoder')
```

It should be `encoder`.